### PR TITLE
[PPD-114] retry reporting service

### PR DESCRIPTION
**Pull request by jacopocarlini**
depends on https://github.com/pagopa/pagopa-infra/pull/230

#### List of Changes

- added GitGuardian in pre-commit
- added retry to reporting-service function

#### Motivation and Context

add a retry policy vs GPD

#### How Has This Been Tested?

- manually
- junits

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: reporting-service/src/main/java/it/gov/pagopa/reporting/functions/UpdateOption.java [hunk 1]
```diff
@@ -8,6 +8,7 @@
 import it.gov.pagopa.reporting.models.OptionsMessage;
 import it.gov.pagopa.reporting.service.GPDService;
 
+import java.util.UUID;
```
Comment (pasqualespica) [inline reporting-service/src/main/java/it/gov/pagopa/reporting/functions/UpdateOption.java:11]: remove

File: reporting-service/src/main/java/it/gov/pagopa/reporting/service/GPDService.java [hunk 2]
```diff
@@ -24,12 +34,52 @@ public static GPDService getInstance() {
...
+     * @param logger        for logging
+     * @return return true if the report is set correctly
+     */
+    @SneakyThrows
+    public boolean setReport(String idPA, PaymentOption paymentOption, Logger logger) {
+        int maxAttemptsValue = Integer.parseInt(Optional.ofNullable(maxAttemps)
+                .orElse("1")); // for testing
+        int delayValue = Integer.parseInt(Optional.ofNullable(delay)
+                .orElse("0")); // for testing
+        for (int i = 1; i <= maxAttemptsValue; i++) {
```
Comment (pasqualespica) [inline reporting-service/src/main/java/it/gov/pagopa/reporting/service/GPDService.java:51]: enqueue newly the message
As made for https://github.com/pagopa/pagopa-debt-position/blob/43dff4fc8134f574a68028d75213cbf4c8d657a7/reporting-batch/src/main/java/it/gov/pagopa/reporting/RetrieveFlows.java#L72

**Review by pasqualespica — CHANGES_REQUESTED** [review]

File: reporting-service/src/main/java/it/gov/pagopa/reporting/functions/UpdateOption.java (hunk 1 above)
Comment (jacopocarlini) [reply reporting-service/src/main/java/it/gov/pagopa/reporting/functions/UpdateOption.java]: done

**Review by pasqualespica — APPROVED** [review]
great 💪

**Pull request merged**